wpcom-helper.php

```diff
@@ -67,29 +85,55 @@ function wpcom_amp_get_stats_extras_url() {
 	return $url;
 }
 
-add_action( 'pre_amp_render', function() {
-	add_filter( 'post_flair_disable', '__return_true', 99 );
-	remove_filter( 'the_title', 'widont' );
+add_action( 'pre_amp_render', 'wpcom_helper_pre_amp_render' );
+
+function wpcom_helper_pre_amp_render() {
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) { // WP.com only.
+		add_filter( 'post_flair_disable', '__return_true', 99 );
+
+		remove_filter( 'the_title', 'widont' );
+
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'filter' ), 11 );
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'maybe_create_links' ), 100 );
+	} else { // Jetpack only.
```

Comment:
I relied on the `else` since this file was only included when `JETPACK__VERSION` is defined, which is commonly shared between both platforms (WP.com has the `JETPACK__VERSION` set to `wpcom`). As far as I could tell, there was no usage if checking that specifically, so defaulted to the typical wp.com check to reduce likelihood of a change breaking things (as parts of the shared codebase assumes the constant is set).

For the check to determine if it is Jetpack, we can check for `JETPACK__VERSION != 'wpcom'. Initially, I was going to opt for `class_exists( 'Jetpack' )`; however, that would return true on wp.com as well. Any particular thoughts or ideas?